model/OAuthClient.php

```diff
@@ -207,7 +207,8 @@ protected function getResponse(RequestInterface $request, $options = [])
     protected function requestAccessToken($params = [])
     {
         $params = $this->addTokenRequestParameters($params);
-        $accessToken = $this->getProvider()->getAccessToken(self::DEFAULT_GRANT_TYPE, $params);
+        $grantType = !empty($this->getOption('grant_type')) ? $this->getOption('grant_type') : self::DEFAULT_GRANT_TYPE;
```

Comment:
This change should be tested. This means that you should modify `OAuthClientTest`. you could make some refactoring in this class maybe?

Reply:
Fixed. https://github.com/oat-sa/extension-tao-oauth/pull/42/files#diff-ac10a0bf21843ed85cdb4aeeba4f2590R156